shared/firebase/service/auth.js

```diff
@@ -1,48 +1,164 @@
-import { fireStorage, auth, google } from "../firebase";
-
-export const register = (credentials, navigation) => {
-  auth
-    .createUserWithEmailAndPassword(credentials.email, credentials.password)
-    .then(result => {
-      console.log(result);
-      navigation.navigate("Login");
-    })
-    .catch(error => {
-      console.log(error);
-    });
-};
+import { fireStore, auth } from "../firebase";
+import * as Google from "expo-google-app-auth";
+import { ANDROID_CLIENT_ID, IOS_CLIENT_ID } from "react-native-dotenv";
+import * as firebase from "firebase";
+import { call, put } from "redux-saga/effects";
+import * as actions from "../../../store/actions/actions";
+
+const FIREBASE_USERS_PATH = "USERS";
 
-export const login = (credentials, navigation) => {
-  auth
-    .signInWithEmailAndPassword(credentials.email, credentials.password)
-    .then(result => {
-      console.log(result);
+export const checkIfLoggedIn = navigation => {
+  auth.onAuthStateChanged(user => {
+    if (user) {
       navigation.navigate("Loading");
-    })
-    .catch(error => {
-      console.log(error);
-    });
+    } else {
+      navigation.navigate("Auth");
+    }
+  });
 };
 
-export const loginByGoogle = navigation => {
-  auth
-    .signInWithPopup(google)
-    .then(result => {
-      console.log(result.user);
-      navigation.navigate("Loading");
-    })
-    .catch(error => {
-      console.log(error);
-    });
+export function* register(action) {
+  const { email, password, firstName, lastName } = action.payload;
+
+  try {
+    const result = yield call(() =>
+      auth.createUserWithEmailAndPassword(email, password)
+    );
+    const createdAt = Date.now();
+    const photoUrl = null;
+    const emailVerified = false;
+    const phoneNumber = null;
+    console.log(result);
+    fireStore
+      .collection(FIREBASE_USERS_PATH)
+      .doc(result.user.uid)
+      .set({
+        email,
+        firstName,
+        lastName,
+        createdAt,
+        photoUrl,
+        phoneNumber,
+        emailVerified,
+      })
+      .then(() => {
+        put(
+          actions.registerSuccess({
+            email,
+            firstName,
+            lastName,
+            phoneNumber,
+            photoUrl,
+          })
+        );
+      });
+  } catch (error) {
+    yield put(actions.registerFailed(error));
+    console.log(error);
+  }
+}
+
+export function* loginByPass(action) {
+  const { email, password } = action.payload;
+  try {
+    const result = yield call(() =>
+      auth.signInWithEmailAndPassword(email, password)
+    );
+    const lastLoginAt = Date.now();
+    console.log(result);
+    fireStore
```

Comment:
Use `yield call` instead of `then`. `call` is blocking so you don't need to worry about async stuff